### DISCUSSION TITLE
It looks like the ajax requests are missing some parameters

### DISCUSSION THREAD

**Discussion by ionutbaltariu** [Sau1707/ModernBot#28] [General]
Hello! Whenever you do some actions in the browser, there are two parameters, **h** and **_** which also send some data to the back-end server. **h** seems to be an md5 hash of something, and **_** is the timestamp of the action. Do you have an idea of the consequences of not sending these parameters? (while using the bot)

**Comment by ionutbaltariu** [comment]
It seems like the **h** parameter is a **csrfToken** that can be retrieved from the **window.Game** object.

**Comment by ionutbaltariu** [comment]
Never mind. I think that the gpAjax that you're using does all of that work.

**Comment by Sau1707** [comment]
Hi,

Yeah the one that I'm using send automatically all the informations that the grepolis back end need to authenticate the user.

Not sending the h token would result in and error, it's used to prevent cross origin attacks and it's checked on every request

Let me know if you have other questions ;)
